### PR TITLE
Add FiraCode@1.204 w/ git auto-update

### LINKED ISSUE #11320: [Request] Add Fira Code
**Library name:** Fira Code
**Git repository url:** https://github.com/tonsky/FiraCode.git
**npm package name or url:** [firacode](https://www.npmjs.com/package/firacode)
**License:** [SIL Open Font License 1.1](https://github.com/tonsky/FiraCode/blob/master/LICENSE)
**Official homepage:** https://github.com/tonsky/FiraCode

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/45547662-request-add-fira-code?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by dakshshah96**
close #11320, cc @tonsky

Pull request for issue: #11320 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/tonsky/FiraCode
 * Official website (optional, not the repository): https://github.com/tonsky/FiraCode/wiki
 * NPM package url (optional):
 * License and its reference: OFL-1.1
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 368
   - Count of stars: 13,722
   - Count of forks: 387
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/FiraCode/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "FiraCode",
+  "version": "1.204",
+  "filename": "fira_code.min.css",
+  "description": "Fira Code is an extension of the Fira Mono font containing a set of ligatures for common programming multi-character combinations.",
+  "homepage": "https://github.com/tonsky/FiraCode/wiki",
+  "keywords": [
+    "font",
+    "monospaces",
+    "programming"
```
Comment (sufuf3) [inline ajax/libs/FiraCode/package.json]: Let's add more keywords `programming-ligatures`, `ligatures`, let it be searched easier.
These keywords show at the blue part.
image

**Comment by dakshshah96** [comment]
@sufuf3 Done 👍

**Comment by sufuf3** [comment]
@extend1994 Could we add old versions (v0.1 ~ v1.102) after this PR merge?
Because it is a high priority.

**Comment by extend1994** [comment]
Maybe we can help add them in this PR? It takes few time. @dakshshah96 Do you mind that we add v0.1 ~ v1.102 for you? cc @sufuf3

**Comment by sufuf3** [comment]
@extend1994 Could we add old versions (v0.1 ~ v1.102) after this PR merge?
Because it is a high priority.
And this PR is stuck a little bit long.
Thanks.

File: ajax/libs/FiraCode/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "autoupdate": {
+    "source": "git",
+    "target": "https://github.com/tonsky/FiraCode.git",
+    "fileMap": [
+      {
+        "basePath": "distr",
+        "files": [
+          "**/*.!(html)"
+        ]
+      }
```
Comment (sashberd) [inline ajax/libs/FiraCode/package.json]: Why not to add next pattern for old versions:
 {
         "basePath": "",
         "files": [
           "FiraCode-*"
         ]
       }
And add only 0.1 version manually
What do you think?

Comment (sufuf3) [reply ajax/libs/FiraCode/package.json]: @sashberd As maintainer said in [this PR](https://github.com/cdnjs/cdnjs/pull/11229#discussion_r145956817)
> if the non-dist basePath part is for the old version only, I won't suggest to use the flexible pattern, just use more static one, we don't want to have unexpected behavior in future.

So, I think It's not necessary to add root directory as the basePath just add old versions.
What do you think?

Thanks.

Comment (sashberd) [reply ajax/libs/FiraCode/package.json]: OK

**Comment by extend1994** [comment]
Both fine, just remember to add them.

**Comment by sufuf3** [comment]
@extend1994 @sashberd So, does this PR looks good to you, except the old versions?
Thanks

File: ajax/libs/FiraCode/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "FiraCode",
+  "version": "1.204",
+  "filename": "fira_code.min.css",
+  "description": "Fira Code is an extension of the Fira Mono font containing a set of ligatures for common programming multi-character combinations.",
+  "homepage": "https://github.com/tonsky/FiraCode/wiki",
```
Comment (extend1994) [inline ajax/libs/FiraCode/package.json]: Need to remove due to the similarity of git repository?

**Comment by sufuf3** [comment]
@dakshshah96
thanks for contribution.
Would you mind working on this and helping improve this PR?
If you have any question, please let us know.

Many thanks.

**Comment by sashberd** [comment]
@sufuf3 @extend1994 @PeterDaveHello I cannot rebase this PR to latest master. Can you help me with it or just close it

**Comment by extend1994** [comment]
@sashberd Maybe try again?

**Comment by sashberd** [comment]
@extend1994 I had tried, but no luck. Can you try to rebase this lib to lates master?

**Comment by extend1994** [comment]
@PeterDaveHello Can we use another pull request for the library? Though I can rebase this one, it takes a lot of time because the branch hasn't been updated since almost one year ago.

**Comment by PeterDaveHello** [comment]
@extend1994 you make the call. I can help rebase also.

**Comment by sashberd** [comment]
@extend1994 @PeterDaveHello I could fetch this one, made a changes and push PR from my fork and then close this one? What do you think?

**Comment by PeterDaveHello** [comment]
I just rebased.

**Comment by sashberd** [comment]
@sufuf3 @extend1994 The PR was updated. FYR

**Comment by sashberd** [comment]
@extend1994 @PeterDaveHello It is again make me a problem on rebaseing... I suggest in such cases to close PR`s and reopen them from master again.
@extend1994 As I told you before, if the problem only in commit, I think message you can do it by yourself.

**Comment by PeterDaveHello** [comment]
@sashberd datails of the problem?

**Comment by sashberd** [comment]
@PeterDaveHello It is on generating patches over 1 hour, like on print screen before. Would you like new print screen?

**Comment by PeterDaveHello** [comment]
@sashberd no screenshot needed, thanks. I just wonder if it's because of how old the branch it is?

**Comment by PeterDaveHello** [comment]
@sashberd Let's do a smal experiment, what do you think?
I'll update the branch first, then would you please try again after that? Thanks.

**Comment by sashberd** [comment]
@PeterDaveHello But you did a rebase, so what would you like to me do from my side?

**Comment by PeterDaveHello** [comment]
@sashberd I'm not sure the reason behind the hanging on your side, so just tried to rebase here ;)

**Comment by sashberd** [comment]
@extend1994 @PeterDaveHello Did an update to this PR. FYR

**Comment by PeterDaveHello** [comment]
@sashberd I think there are some small changes need to be made?

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Pull request closed without merging**